board/app/models/user.php

```diff
@@ -103,37 +142,83 @@ public function login()
             'username' => $this->username,
             'password' => md5($this->password)
         );
+        $user = $db->row("SELECT id, username FROM user WHERE BINARY username = :username AND BINARY password = :password", $params);
 
-    $user = $db->row("SELECT id, username FROM user WHERE BINARY username = :username && BINARY password = :password", $params);
-
-        if(!$user) 
-        {
-            $this->is_validated = false; 
+        if(!$user) {
             throw new RecordNotFoundException('No Record Found');
         }
 
-    $_SESSION['user_id'] = $user['id'];
-    $_SESSION['username'] = $user['username'];
+        $_SESSION['user_id'] = $user['id'];
+        $_SESSION['username'] = $user['username'];
     }
 
-    public function is_password_match()
+    public function update()
     {
-        return $this->password == $this->confirm_password;
+        if (!$this->validate()) {
+            throw new ValidationException('Invalid user credentials');
+        }
+
+        try {
+            $db = DB::conn();
+            $db->begin();
+            $db->update(
+                'user', array(
+                    'username' => $this->username,
+                    'first_name' => $this->first_name,
+                    'last_name' => $this->last_name,
+                    'password' => $this->password
+                    ),
+                array('id' =>$this->user_id)
+                );
```

Comment:
Done po. Thank you :bow:
--> https://github.com/maragatchalian/FoodFacts/commit/7c09ebdeed3397649e9e4d9f85406ef19f3230a6